--- a/libalpm/scripts/mkinitcpio	2023-03-19 08:24:59.000000000 -0600
+++ b/libalpm/scripts/mkinitcpio	2023-03-21 15:46:00.172629798 -0600
@@ -8,7 +8,7 @@
 
 process_preset() {
     if [[ -n "$pkgbase" && -e "$preset" ]]; then
-        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset); then
             if [[ ! -e "$preset.pacsave" ]]; then
                 # save the preset as pacsave
                 mv -- "$preset" "$preset.pacsave" && return 0
@@ -76,7 +76,7 @@
             mv -- "${preset}.pacsave" "$preset"
         else
             # create the preset from the template
-            sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset \
+            sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset \
                 | install -Dm644 /dev/stdin "$preset"
         fi
     fi
@@ -130,6 +130,11 @@
         continue
     fi
 
+    if ! read -r kernelbase > /dev/null 2>&1 < "${line%/vmlinuz}/kernelbase"; then
+        # this kernel has no kernelbase, use pkgbase
+        kernelbase="${pkgbase}"
+    fi
+
     case "$1" in
         install) install_kernel "$pkgbase";;
         remove) remove_kernel "$pkgbase";;
