--- a/functions	2024-01-15 22:53:37.000000000 +0700
+++ b/functions	2024-01-16 10:17:38.390700191 +0700
@@ -732,7 +732,8 @@ add_file() {
 
     # If the parent directory is a symlink, add the directory symlink.
     # What if there is a symlink higher up in the file path, you ask? We simply hope that never happens.
-    if [[ -z "$mode" && "$src" == "$dest" && -L "${dest%/*}" && ! -e "${BUILDROOT}${dest%/*}" ]]; then
+    if [[ -z "$mode" && "$src" == "$dest" && -L "${dest%/*}" && ! "${dest%/*}" =~ "extramodules" \
+    && ! -e "${BUILDROOT}${dest%/*}" ]]; then
         add_full_dir "$(realpath -- "${dest%/*}")"
         add_symlink "${dest%/*}"
         [[ -e "$(realpath -- "${BUILDROOT}${dest}")" ]] && return
@@ -752,6 +753,7 @@ add_file() {
 
     # unlike install, cp does not create directories leading to the destination
     [[ -z "$mode" && ! -e "${BUILDROOT}${dest%/*}" ]] && add_dir "${dest%/*}"
+    [[ -z "$mode" && "${dest%/*}" =~ "extramodules" ]] && add_dir "${dest%/*}"
 
     if [[ -e "${BUILDROOT}${dest}" ]]; then
         quiet 'overwriting file: %s' "$dest"
