@@ -856,7 +856,7 @@ if [[ -n "$_optgenimg" ]]; then
         die "Unable to write to '%s'" "$_optgenimg"
     fi
 
-    _optcompress="${_optcompress:-"${COMPRESSION:-zstd}"}"
+    _optcompress="${_optcompress:-"${COMPRESSION:-gzip}"}"
     if ! type -P "$_optcompress" >/dev/null; then
         warning "Unable to locate compression method: '%s'" "$_optcompress"
         _optcompress='cat'
--- b/mkinitcpio.conf
+++ a/mkinitcpio.conf
@@ -52,15 +52,15 @@
 HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
 
 # COMPRESSION
+# Use this to compress the initramfs image. By default, gzip compression
-# Use this to compress the initramfs image. By default, zstd compression
 # is used. Use 'cat' to create an uncompressed image.
-#COMPRESSION="zstd"
 #COMPRESSION="gzip"
 #COMPRESSION="bzip2"
 #COMPRESSION="lzma"
 #COMPRESSION="xz"
 #COMPRESSION="lzop"
 #COMPRESSION="lz4"
+#COMPRESSION="zstd"
 
 # COMPRESSION_OPTIONS
 # Additional options for the compressor
